core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1782,4 +1782,16 @@ class ParserTest extends munit.ScalaCheckSuite {
       assertEquals(left.parse(str), right.parse(str))
     }
   }
+
+  property("Or parser works as expected") {
```

Comment:
Can we also test the method version on Parser0?

Reply:
Good point!